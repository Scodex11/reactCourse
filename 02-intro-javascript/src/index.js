const nombre  = "Facundo";
const apellido = "Garcia";

const nombreCompleto = `${nombre} ${apellido}`;

console.log(nombreCompleto);

function getSaludo(nombre) {
    return 'Hola ' + nombre;
}

console.log(`Este es un texto: ${getSaludo(nombreCompleto)}`);