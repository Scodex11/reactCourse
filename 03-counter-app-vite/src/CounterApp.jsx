import { useState } from "react";
import PropTypes from "prop-types"; //Para los tipos de las PROPS

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value); // HOOK para manejar el estado de la variable

  const handleAdd = () => {
    //setCounter(counter + 1);
    setCounter((c) => c + 1); //Es lo mismo que la linea anterior, pero sin usar el 'counter'
  };

  const handleSubs = () => {
    setCounter((c) => c - 1); 
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2> {/*Para manejar los cambios se utiliza el 'counter'*/}
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubs}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 0,
};
