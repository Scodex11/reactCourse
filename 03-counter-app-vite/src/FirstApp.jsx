import PropTypes from 'prop-types'; //Para los tipos de las PROPS


export const FirstApp = ({name, title, subTitle}) => {
   return (
    <>
        <h1>{title}</h1>
        <p>{subTitle}</p>
        <p>{name}</p>
    </>
  )
}


//Deben estar al final por ESTANDARIZACIÃ“N

FirstApp.propTypes = {
    name: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Sin nombre',
    subTitle: 'Sin subtitulo',
    title: 'Sin asdasdasdasdds',
}

