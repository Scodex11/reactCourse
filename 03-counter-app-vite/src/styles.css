html, body{
    background-color: #383636;
    color: #FFFFFF;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem; /*what a heall it is*/
    padding: 70px;
}