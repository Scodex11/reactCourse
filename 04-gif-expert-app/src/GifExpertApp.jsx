import { useState } from 'react'
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const arrayCategories = ['Dragon Ball']

    const [categories, setCategories] = useState(arrayCategories)
 
    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)){
            alert("¡La categoría ya existe!");
            return;
        } //No se permite agregar categorias repetidas

        setCategories([newCategory, ...categories])
    }


    return (
    <>
        <h1>GifExpertApp</h1>

        {/* INPUT */}
        <AddCategory onNewCategory = { onAddCategory } />    
    
       
        {
            categories.map( category => (
                <GifGrid 
                    key = { category }
                    category = { category }
                />
            ))
        }


    </>
  )
}
