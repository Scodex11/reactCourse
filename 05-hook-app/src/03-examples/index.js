export * from './Loading';
export * from './Card';
export * from './MultipleCustomHooks';