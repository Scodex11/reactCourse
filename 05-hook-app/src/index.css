body{
  padding: 20px;
  background-color: #90bd90;
}

button{
  margin-right: 10px;
}

.card-container {
  text-align: center;
  padding: 20px;
  background-color: #f0f0f0;
  border-radius: 10px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}

.card-container img {
  display: block;
  margin: 0 auto;
  border-radius: 50%;
  max-width: 200px;
}

.card-container blockquote {
  margin-top: 20px;
}