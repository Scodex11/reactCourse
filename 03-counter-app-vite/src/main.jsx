import React from 'react'
import ReactDOM from 'react-dom'
import { CounterApp } from './CounterApp'
import './styles.css' 
import { FirstApp } from './FirstApp';
 
//Renderizar nuestra app
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title='Hola soy un titulo'/>
    </React.StrictMode>
);