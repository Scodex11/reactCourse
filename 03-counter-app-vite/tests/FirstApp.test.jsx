import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas de FirstApp<>", () => {
  
  
    test("Debe hacer MATCH con el SNAPSHOT", () => {
        const title = "Hola soy un titulo";

        //El render lo que hace es crear un FIRSTAPP  pero PROPIO del TEST -> Por lo que SI MODIFICAS las props DEL ORIGINAL no cambia en nada
        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();
    });

    test('EL title debe estar en un h1', () => { 
        
        const title = "Sin titulo";
        const { container, getByText } = render(<FirstApp title={title} />);

        /*Pruebas*/ 
        expect(getByText(title)).toBeTruthy(); //Si existe el elemento con el texto 'title' = "Sin titulo"
        
        const h1 = container.querySelector('h1'); //Guardamos el primer elemento del FirstApp tipo h1
        
        expect(h1.innerHTML).toContain(title); //Si el innerHTML del h1 contiene el title = "Sin titulo"


     })

});
