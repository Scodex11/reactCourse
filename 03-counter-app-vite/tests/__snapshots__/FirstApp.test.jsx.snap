// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de FirstApp<> Debe hacer MATCH con el SNAPSHOT 1`] = `
<div>
  <h1>
    Hola soy un titulo
  </h1>
  <p>
    Sin subtitulo
  </p>
  <p>
    Sin nombre
  </p>
</div>
`;
